src/utils/file-type.ts

```diff
@@ -0,0 +1,28 @@
+import { pipe } from "fp-ts/lib/function";
+import * as RA from "fp-ts/ReadonlyArray";
+import { match } from "ts-pattern";
+
+export type FileType = "pdf";
+
+/**
+ * Verify if the input buffer contains a PDF using the magic number in the first bytes (see https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files)
+ *
+ * @param data a binary buffer
+ * @returns true if is a pdf
+ */
+export const isPdf = (data: Buffer) => data.toString("binary", 0, 4) === "%PDF";
+
+export const typeToCheck = (type: FileType) =>
+  match(type)
+    .with("pdf", () => isPdf)
+    .exhaustive();
+
+export const getIsFileTypeForTypes = (types: ReadonlyArray<FileType>) => (
+  data: Buffer
+) =>
+  pipe(
+    types,
+    RA.map(typeToCheck),
+    RA.map(is => is(data)),
+    RA.reduce(false, (current, next) => current || next)
```

Comment:
This will return `true` if any value from the second element to the last is `false`. Maybe is better use the follow with a `NonEmptyArray<FileType>`
```suggestion
    RA.every(identity)
```

Reply:
Nice suggestion!
I think we have to use `RA.some(identity)` instead of `RA.every(identity)`, becouse we want to match at least one of the allowed types.
I've refactored it and added a unit test.
Thank you!

Reply:
You are right, thanks. So we can improve a little bit the method signature. Pls add even a small documentation for the method.
```typescript
export const getIsFileTypeForTypes = (types: ReadonlySet<FileType>) => (
  data: Buffer
) =>
  pipe(
    Array.from(types),
```

Reply:
Nice, and using fp-ts ReaonlySet will prevent us to call the same file type checker two times.
I've replaced the method signature and added a small doc.